File: POO/Ejercicio3/Codigo/Coche.py
# 3. Crea una clase Coche con marca, modelo y velocidad
# a)
# Agrega un método acelerar () que aumente la velocidad en 10
# b)
# Agrega un método frenar () que disminuya la velocidad en 5
# c)
# Crea dos coches, aceléralos, frénalos y muestra sus velocidades

class Coche:
    def __init__(self, marca, modelo, velocidad):
        self.marca = marca
        self.modelo = modelo
        self.velocidad = velocidad

    def acelerar(self):
        self.velocidad += 10
        print(f"La velocidad es: {self.velocidad}")

    def frenar(self):
        self.velocidad -= 5
        print(f"La velocidad es: {self.velocidad}")

    def mostrar(self):
        print(f"El auto es: {self.marca}")
        print(f"El modelo es: {self.modelo}")
        print(f"Su velocidad es: {self.velocidad}")


if __name__ == "__main__":
    c1 = Coche("Hyundai", "Creta 2019", 190)
    c2 = Coche("Toyota", "Toyota Hilux", 175)

    print("---------LOS AUTOS SON--------------")
    c1.mostrar()
    c2.mostrar()

    print("----------AUTOS ACELERADOS----------")
    c1.acelerar()
    c2.acelerar()

    print("----------AUTOS FRENADOS------------")
    c1.frenar()
    c2.frenar()

    print("---------Velocidades--------------")
    c1.mostrar()
    c2.mostrar()

File: POO/Ejercicio3/Coche.java
package POO.Ejercicio3;

public class Coche {
    private String marca;
    private String modelo;
    private int velocidad;

    public Coche(String marca, String modelo, int velocidad) {
        this.marca = marca;
        this.modelo = modelo;
        this.velocidad = velocidad;
    }

    public void Acelerar(){
        this.velocidad = this.velocidad + 10;
        System.out.println("La velocidad es " + this.velocidad);

    }

    public void Frenar(){
        this.velocidad = this.velocidad - 5;
        System.out.println("La velocidad es " + this.velocidad);
    }

    public void mostrar(){
        System.out.println("El auto es " + this.marca);
        System.out.println("El modelo es " + this.modelo);
        System.out.println("Su velocidad es " + this.velocidad);
    }

    public static void main(String[] args) {
        Coche coche1 = new Coche("Honda", "Civic", 200);
        Coche coche2 = new Coche("Toyota", "Corolla", 100);

        System.out.println("COCHES");

        coche1.mostrar();
        coche2.mostrar();

        System.out.println("------------------------------------");

        System.out.println("Coches Acelerados:");
        coche1.Acelerar();
        coche2.Acelerar();
        System.out.println("--------------------------------------");
        System.out.println("Coches Frenados:");
        coche1.Frenar();
        coche2.Frenar();
        System.out.println("--------------------------------------");
        System.out.println("Velocidades:");
        coche1.mostrar();
        coche2.mostrar();
    }
}
